Index: cloud-init-0.7.6-bzr1060/systemd/cloud-config.service
===================================================================
--- cloud-init-0.7.6-bzr1060.orig/systemd/cloud-config.service
+++ cloud-init-0.7.6-bzr1060/systemd/cloud-config.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Apply the settings specified in cloud-config
-After=network.target syslog.target cloud-config.target
+After=network-online.target syslog.target cloud-config.target
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
Index: cloud-init-0.7.6-bzr1060/systemd/cloud-init.service
===================================================================
--- cloud-init-0.7.6-bzr1060.orig/systemd/cloud-init.service
+++ cloud-init-0.7.6-bzr1060/systemd/cloud-init.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-After=local-fs.target network.target cloud-init-local.service
+After=local-fs.target network-online.target cloud-init-local.service
 Before=sshd.service sshd-keygen.service systemd-user-sessions.service
-Requires=network.target
+Requires=network-online.target
 Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
 
 [Service]
Index: cloud-init-0.7.6-bzr1060/systemd/cloud-final.service
===================================================================
--- cloud-init-0.7.6-bzr1060.orig/systemd/cloud-final.service
+++ cloud-init-0.7.6-bzr1060/systemd/cloud-final.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Execute cloud user/final scripts
-After=network.target syslog.target cloud-config.service rc-local.service
+After=network-online.target syslog.target cloud-config.service rc-local.service
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
