Index: cloud-init-0.7.6/systemd/cloud-init.service
===================================================================
--- cloud-init-0.7.6.orig/systemd/cloud-init.service
+++ cloud-init-0.7.6/systemd/cloud-init.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
 After=local-fs.target network-online.target cloud-init-local.service
-Before=sshd.service sshd-keygen.service
+Before=sshd.service sshd-keygen.service systemd-user-sessions.service
 Requires=network-online.target
 Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
 
